src/file_io.c

```
@@ -1,309 +1,309 @@
+	{
+		printf("\nError: Memory allocation failed.\n");
+		fclose(file);
+		return NULL;
+	}
+
+	char *line = malloc(10000 * sizeof(char)); // Large enough buffer for long rows
+	if (!line)
+	{
+		printf("\nError: Memory allocation for line buffer failed.\n");
+		fclose(file);
+		return NULL;
```

Cppcheck (2.10-2)

[error] Memory leak

(memleak)